pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -187,6 +174,22 @@ private static void validateGroupByClause(PinotQuery pinotQuery)
     }
   }
 
+  private static boolean expressionOutsideGroupByList(Expression expr, Set<Expression> groupByExprs) {
+    if (expr.getType() == ExpressionType.LITERAL || isAggregateExpression(expr) || groupByExprs.contains(expr)) {
```

Comment:
Will this `groupByExprs.contains(expr)` handles functions or just for identifiers?

Reply:
Hi @fx19880617, it will handle all expression types (including nested). Expression overrides equals() & hashCode() so this should work for arbitrary Expression

Reply:
👍